### PR TITLE
Reducers and actions

### PR DESCRIPTION
**Pull request by thankgodr**
This PR introduces

- Book Reducer
- Book Actions
- Category Reducers
- Category Actions
- single store for state management
- Code Linted

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import categoryReducer from './categories/categories';
+
+const rootReducer = combineReducers({
+  books: bookReducer,
+  category: categoryReducer,
+});
+
+const store = configureStore({
+  reducer: rootReducer,
+});
```
Comment (omar-muhamad) [inline src/redux/configureStore.js]: - [x] [Optional] When you import `configureStore` from `redux toolkit` you don't need `combineReducer` you can add all reducers like that:
```javascript
const store = configureStore({
  reducer: {
     books: bookReducer,
     category: categoryReducer,
}});
```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+export const BOOK_ACTION = {
+  ADD: 'add_book',
+  REMOVE: 'remove_book',
+};
+
```
Comment (omar-muhamad) [inline src/redux/books/books.js]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#:~:text=Define%20action%20types%20for%20adding%20and%20removing%20a%20book.), Please Define action types for adding and removing a book using [ducks pattern](https://github.com/erikras/ducks-modular-redux).
[Hint]: For example: `const ADD_BOOK = bookstore/books/ADD_BOOK`

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export const CATEGORY_ACTION = {
+  STATUS: 'category_status',
+};
+
```
Comment (omar-muhamad) [inline src/redux/categories/categories.js]: - [x] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#:~:text=Define%20an%20action%20type%20for%20checking%20the%20status.), Please define an action type for checking the status using [ducks pattern](https://github.com/erikras/ducks-modular-redux).

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+export default function categoryReducer(state = initialState, action) {
+  switch (action.type) {
+    case CATEGORY_ACTION.STATUS:
+      return 'Under construction';
+    default:
+      return state;
+  }
+}
```
Comment (omar-muhamad) [inline src/redux/categories/categories.js]: - [x] [Optional] You using the old way to create a redux reducer in the `redux toolkit` you suppose to use `createSlice` to [create a Redux State Slice](https://redux-toolkit.js.org/tutorials/quick-start#create-a-redux-state-slice).

File: src/redux/books/books.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    case BOOK_ACTION.REMOVE:
+      state = {
+        ...state,
+        books: state.books.filter((book) => book.id !== action.payload.bookId),
+      };
+      return state;
+    default:
+      return state;
+  }
+}
```
Comment (omar-muhamad) [inline src/redux/books/books.js]: - [x] [Optional] Same here, You using the old way to create a redux reducer in the `redux toolkit` you suppose to use `createSlice` to [create a Redux State Slice](https://redux-toolkit.js.org/tutorials/quick-start#create-a-redux-state-slice).

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @thankgodr,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @thankgodr, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**